### ISSUE TITLE
Apollo Query and Mutation components: JSX props should not use functions (react/jsx-no-bind)

### ISSUE THREAD

**Issue by johnunclesam** [yannickcr/eslint-plugin-react#1872]
I'm using the Apollo's `<Mutation>` component like this:

```js
import gql from "graphql-tag";
import { Mutation } from "react-apollo";

const ADD_TODO = gql`
  mutation addTodo($type: String!) {
    addTodo(type: $type) {
      id
      type
    }
  }
`;

const AddTodo = () => {
  let input;

  return (
    <Mutation mutation={ADD_TODO}>
      {(addTodo, { data }) => (
        <div>
          <form
            onSubmit={e => {
              e.preventDefault();
              addTodo({ variables: { type: input.value } });
              input.value = "";
            }}
          >
            <input
              ref={node => {
                input = node;
              }}
            />
            <button type="submit">Add Todo</button>
          </form>
        </div>
      )}
    </Mutation>
  );
};
```

I get this error:

`[eslint] JSX props should not use functions (react/jsx-no-bind)`

I think it is complaining about:

```js
onSubmit={e => {
  e.preventDefault();
  addTodo({ variables: { type: input.value } });
  input.value = "";
}}
```

because is an arrow function.

But it's the default in Apollo React Docs: https://www.apollographql.com/docs/react/essentials/mutations.html.

So... it's a good way or we need a function in constructor?

How can I fix this and suggest it to Apollo React team?

**Comment by ljharb** [comment]
To satisfy the rule, you either need to configure it to allow inline arrow functions, or you need a constructor-bound instance method.

**Comment by ljharb** [comment]
In general, it's a minor performance hazard. In this case, it's on a DOM component, so it's fine.

**Comment by johnunclesam** [comment]
Can you explain it better this phrase? Thanks.

> In this case, it's on a DOM component, so it's fine.

**Comment by ljharb** [comment]
Yes - DOM/html components (ie, host nodes, like `div`, `span`, etc) don't have the same pitfalls here that custom components do. Namely, they're not PureComponents, so there can't possibly be any rendering performance hit (only the minimal hit of "creating new functions")
